lib/puppet/type/cisco_upgrade.rb

```diff
@@ -0,0 +1,123 @@
+#
+# Manages the version of Cisco Image running on a device.
+#
+# Copyright (c) 2017 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_upgrade) do
+  @doc = "Manages the version of Cisco Image running on a device.
+
+  ```
+  cisco_upgrade {\"<instance_name>\":
+    ..attributes..
+  }
+  ```
+
+  There can only be one instance of cisco_upgrade i.e. 'image'
+
+  Example:
+  ```
+    cisco_upgrade {'image' :
+      version           => '7.0(3)I5(1)'
```

Comment:
Add a `,` at the end of the version line.  I cut and pasted this into my manifest and found it :)